src/test/test-namespace.c

```diff
@@ -93,6 +94,11 @@ static void test_shareable_ns(unsigned long nsflag) {
                 return;
         }
 
+        /* If we are running inside LXC and with AppArmor, most likely creating a new network namespace
+         * is going to fail, so just skip immediately. */
+        if (detect_container() == VIRTUALIZATION_LXC && mac_apparmor_use())
+                return (void) log_tests_skipped("in LXC container with Apparmor");
```

Comment:
this seems like a hack. why would that fail? that seems like a policy issue, not an lxc/apparmor issue?

Reply:
The default apparmor policy denies network namespace creation inside lxc containers

Reply:
with EAGAIN?

Reply:
do you have a link to that policy?

Reply:
https://sources.debian.org/src/lxc/1%3A5.0.2-1%2Bdeb12u2/config/apparmor/

Reply:
where does it say anything about namespaces?

and where is the EAGAIN in there? can't see it?

Reply:
I am very much not an apparmor expert, but I think you don't define the errors on denials, it just returns what it returns, I don't see any specific errors defined on my machine in /etc/apparmor.d
And judging from a quick google, I think the restrictions are applied to sysfs and mount calls, but might be wrong on these details

Reply:
Also this unit test change doesn't check for any error, just for the initial condition, like many other tests do (and indeed this very same one, which checks for UID 0 just above this check). So this change has nothing to do with eagain.

Reply:
hmm, i'd prefer if we'd instead use have_namespaces() here. (which currently checks CLONE_NEWNS, but should be trivial to update to test for CLONE_NEWNET instead).

i.e. i dislike tests for explicit software, and much prefer generic checks.

Reply:
The problem is apparmor, not the namespace, it's the same denial on flock that makes this fail. We have tests checking for specific environments already, for unit tests it's unavoidable as they also run at build time in the weirdest setups

Reply:
i see, but i still don't like this, as it means we'll never run these tests anymore, even when lxc/aa gets fixed.

there must be some way to just skip this test on debian CI for now?

Reply:
afaik we do not carry such explicit workarounds for bugs in other software anywhere in the tree, i doubt we should add this here.

Reply:
We do carry similar checks elsewhere, again only in tests. And given it's a unit test I cannot skip it individually, only the whole unit tests run which would be silly. I can change the test to avoid failing on eacces, that is the only alternative - which means a similar regression in other environments will not be noticed, but of that's the only way to get this merged I'll do that

Reply:
Ok, checking for permission errors explicitly now.